### PR TITLE
chore: avoid adding `.build` directory to client's workspace

### LINKED ISSUE #758: Do not add `.build` directory when running `:swift_update_pkgs`
In #747, we added a `.bazelignore` to the `tca_example`. None of the other examples require this. We should understand why this is happening.

Related: Some clients have requested that the `.build` directory not be placed in the workspace. There are several options:

1. Remove the `.build` directory after running `swift_update_pkgs`. The downside is that some clients may want a `.build` directory for the Swift LSP to work properly.
2. Specify `--build-path` when running Swift package resolution and place it elsewhere. Maybe in a temp directory?

### PR DESCRIPTION
**Pull request by cgrindel**
- Pass `--build-path` to select `swift package` calls that are executed during `update-repos` (i.e., `//:swift_update_pkgs`).
- Remove obsolete `.bazelignore` files.
- Remove call to utility that cleans up `.build` directories.

Closes #758.

### CODE REVIEW

**Comment by cgrindel** [comment]
cc: @jpsim @AttilaTheFun

File: gazelle/update_repos.go [hunk 1]
```diff
@@ -43,15 +43,23 @@ func importReposFromPackageManifest(args language.ImportReposArgs) language.Impo
 	sc := swiftcfg.GetSwiftConfig(c)
 	sb := sc.SwiftBin()
 
+	// Create a build directory
+	buildDir, err := os.MkdirTemp("", "rspm_build_dir")
```
Comment (luispadron) [inline gazelle/update_repos.go:47]: Is the `r` intentional? 

```suggestion
	buildDir, err := os.MkdirTemp("", "spm_build_dir")
```

**Review by luispadron — APPROVED** [review]
Sweet this is helpful, we can now remove our `.bazelignore` in our project as well

**Comment by luispadron** [comment]
Would be good as well to release a new version with this change + some of the other stuff we've merged

File: gazelle/update_repos.go (hunk 1 above)
Comment (cgrindel) [reply gazelle/update_repos.go:47]: Yes. It was `rspm` was meant to mean `rules_swift_package_manager`. I wanted to make it clear that it was created by `rules_swift_package_manager`.

**Comment by cgrindel** [comment]
> Sweet this is helpful, we can now remove our .bazelignore in our project as well

Yes. If you run into an issue, please let me know.

**Pull request merged**